content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md

```diff
@@ -0,0 +1,200 @@
+{{% codeblock %}}
+
+Example dockerfile - Template
+
+```dockerfile
+FROM continuumio/miniconda3
+
+RUN conda install jupyter
+RUN pip install <your-packages>
+RUN mkdir /your-env
+```
+
+{{% /codeblock %}}
+
+{{% tip %}}
+If your environment requires you to install a specific version of a package you can use `pip install <package>==<version>` after the `RUN` instruction in the dockerfile instead of the regular `pip install <package>` (e.g. `RUN pip install pytest==7.0.1`)
+{{% /tip %}}
+
+Once your dockerfile is ready, open the command line in the directory where it is located and run:
+
+`docker build -t <your_image_name> .`
+
+{{% warning %}}
+
+Docker builds images from a dockerfile and its context (i.e. files located alongside the dockerfile), thus it is generally recommended to place your dockerfile in a directory dedicated to storing the dockerfile itself and its context to ensure the building process is run smoothly. If you are building an image for an environment where the dockerfile will mostly cover the installation of the right packages and no contextual actions are taking place you could just place it in an empty directory, however, bear in mind this may not always be the case.
+
+{{% /warning %}}
+
+After having done this, Docker will have generated an image (i.e. a model of your environment) from it, which is now ready to be exported! 
+
+This image will be identified by the name you decided to give to it after the flag `-t` as well as by an alphanumerical image ID of 12 characters assigned automatically by docker.
+
+{{% tip %}}
+
+You can see a list of all your images and the details about them by running `docker images` in your command line.
+
+{{% /tip %}}
+
+
+### Pushing your image to Docker Hub
+
+[Docker Hub](https://hub.docker.com/) is the most popular platform for hosting and distributing docker images, which are organized in repositories. One of the advantages of hosting your container image on Docker Hub is that anyone willing to use it (i.e. replicate your environment in this case) will be able to do so remotely simply by pulling from their command line.
+
+{{% tip %}}
+Before you try to push your image make sure you are properly logged in to your Docker Hub account. To do that run `docker login` in the command line and then introduce your Docker Hub username and password.
+{{% /tip %}}
+
+A Docker Hub repository is designed to contain one or multiple versions of a docker image, which are differentiated by their tag. It is important that you assign an appropriate tag to each of your images within a repository so you can appropriately differentiate them. To assign a tag to your recently built image you should type the following in the command line:
```

Comment:
If you don't specify a tag it isn't a huge problem: docker/DockerHub will just assign the `latest` tag automatically to the latest build of the image

Reply:
Fixed, I've added that extra info to the paragraph!